### PR TITLE
Jroach/us129228

### PR DESCRIPTION
**Pull request by jeroach**
[US129228](https://rally1.rallydev.com/#/42960320374d/custom/236803223728?Iteration=%2Fiteration%2F450584616044&Name=%5BDiscussions%5D%20Add%20demo%20page%20and%20vdiff%20test%20for%20new%20d2l-table%20logic&Project=%2Fproject%2F42960320374&Release=u&cpoid=42960320374&detail=%2Fuserstory%2F601864688567&rankScope=%2Fiteration%2F450584616044&rankTo=BOTTOM&typeDef=7362609683&fdp=true?fdp=true): [Discussions] Add demo page and vdiff test for new d2l-table logic

### CODE REVIEW

File: components/left-panel/discussions/consistent-evaluation-discussion-post-score.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import '@brightspace-ui/core/components/icons/icon.js';
+import '@brightspace-ui-labs/grade-result/d2l-grade-result.js';
```
Comment (jeroach) [inline components/left-panel/discussions/consistent-evaluation-discussion-post-score.js:2]: This import was previously missing, and causing issues with the demo page (as it uses this component)

File: demo/consistent-evaluation-evidence-discussion.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+		<meta name="viewport" content="width=device-width, initial-scale=1.0">
+		<meta http-equiv="X-UA-Compatible" content="ie=edge">
+		<meta charset="UTF-8">
+	</head>
+	<body>
+
+		<d2l-demo-page page-title="d2l-consistent-evaluation-evidence-discussion">
+			<h2>Discussion Page</h2>
+			<d2l-demo-snippet>
+				<d2l-consistent-evaluation-evidence-discussion id="discussion-post-with-attachments"
```
Comment (jeroach) [inline demo/consistent-evaluation-evidence-discussion.html:20]: This file is the overall demo page for discussions - it looks new, but I just renamed the other file.

This should likely get fleshed out as well (along with corresponding vdiff tests).

But that can be another story

**Review by maddiegabriel — APPROVED** [review]
LGTM - thanks for the crash course!

**Review by JamieTran — APPROVED** [review]
LGTM, great demo & test coverage 👍

**Comment by JamieTran** [comment]
You mentioned previously that a refactor might need to happen with the discussions `Promise.all()` ~ is that still the case, looks like you found a work-around?

**Comment by jeroach** [comment]
> You mentioned previously that a refactor might need to happen with the discussions `Promise.all()` ~ is that still the case, looks like you found a work-around?

Indeed I found a work-around - I was able to just set the underlying object in the demo/tests

**Pull request merged**